cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java

```diff
@@ -156,42 +218,46 @@ public void prev(HttpRequest request, HttpResponder responder, @PathParam("names
-    try {
-      Filter filter = FilterParser.parse(filterStr);
-      long fromTimeMs = TimeUnit.MILLISECONDS.convert(fromTimeMsParam, TimeUnit.SECONDS);
-      long toTimeMs = TimeUnit.MILLISECONDS.convert(toTimeMsParam, TimeUnit.SECONDS);
+  public void getAdapterLogs(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @PathParam("adapter-id") String adapterId,
+                             // The following 3 query params should be removed,
+                             // when we have a way of querying adapter service
+                             // for the template and program, given an adapter name
+                             @QueryParam("template") String templateId,
+                             @QueryParam("programtype") String programType,
+                             @QueryParam("programid") String programId,
+                             @QueryParam("start") @DefaultValue("-1") long fromTimeMsParam,
+                             @QueryParam("stop") @DefaultValue(Long.MAX_VALUE + "") long toTimeMsParam,
+                             @QueryParam("escape") @DefaultValue("true") boolean escape,
+                             @QueryParam("filter") @DefaultValue("") String filterStr) {
+    LoggingContext loggingContext =
+      LoggingContextHelper.getLoggingContext(namespaceId, templateId, programId,
+                                             getProgramType(ProgramType.valueOfCategoryName(programType)), adapterId);
+    doGetLogs(responder, loggingContext, fromTimeMsParam, toTimeMsParam, escape, filterStr);
+  }
 
-      if (fromTimeMs < 0 || toTimeMs < 0 || toTimeMs <= fromTimeMs) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST, "Invalid time range. 'start' and 'stop' should be " +
-          "greater than zero and 'stop' should be greater than 'start'.");
-        return;
-      }
-      LoggingContext loggingContext =
-        LoggingContextHelper.getLoggingContext(namespaceId, templateId, programId,
-                                               getProgramType(ProgramType.valueOfCategoryName(programType)), adapterId);
-      ChunkedLogReaderCallback logCallback = new ChunkedLogReaderCallback(responder, logPattern, escape);
-      logReader.getLog(loggingContext, fromTimeMs, toTimeMs, filter, logCallback);
-    } catch (SecurityException e) {
-      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
-    } catch (IllegalArgumentException e) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
-    } catch (Throwable e) {
-      LOG.error("Caught exception", e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
-    }
+  @GET
+  @Path("/namespaces/{namespace-id}/adapters/{adapter-id}/runs/{run-id}/logs")
+  public void getAdapterRunIdLogs(HttpRequest request, HttpResponder responder,
```

Comment:
So adapter next and prev aren't part of this PR right? Can we report a jira to track?